mail_optional_follower_notifications/models/mail_message.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class MailMessage(models.Model):
+    _inherit = 'mail.message'
+
+    @api.model
+    def create(self, values):
+        ctx = self.env.context.copy()
+        if not ctx.get('notify_followers') and values.get('partner_ids'):
+            force_partners_to_notify = []
+            for partner_id in values.get('partner_ids'):
+                if isinstance(partner_id, tuple) and\
```

Comment:
can't you better collect them and use https://github.com/OCA/OCB/blob/8.0/openerp/models.py#L5104 ?